Bare bones frontend

Yetalon opened the pull request:
Started work on the barebones frontend created landing page for home and characters.

Yetalon committed:
Created framework fronted

Yetalon committed:
updated navbar created characters landing page

PhiYerion reviewed (changes requested):
Make a new service in ecs that serves this.

Yetalon committed:
script added to read me

Yetalon committed:
moved deploy to top level dir and fixed file paths

PhiYerion commented on frontend/package.json:
```diff
@@ -0,0 +1,28 @@
+  },
+  "dependencies": {
+    "react": "^19.0.0",
+    "react-dom": "^19.0.0"
```
Needs react-router-dom

PhiYerion replied:
React-router-dom just rexports react-router. Use react-router instead (https://www.npmjs.com/package/react-router-dom)

Yetalon committed:
Merge branch 'main' of https://github.com/AppalachianCoding/rpg-app into bare-bones-frontend

Yetalon committed:
created docker file and updated deploy

PhiYerion committed:
Remove image change check (doesn't work) & cleanup

Fix workflows

Delete oil backend/deploy dir

Fix aws region in deploy

Fix docker push order

Add react-router-dom to package.json

Fix unclean merge

Order of operations: backend/deploy/ -> deploy/, main -> frontend (inc.
updated backend/deploy/), backend/deploy -> /dev/null. Then this commit
to main/backend/deploy -> bare-bones-frontend/deploy/

Fix docker push & build

Fix unclean deploy again

PhiYerion committed:
Fix naming conficts

Fix container reference

Fix health check

PhiYerion committed:
map ip for ecs

Fix subnet ranges

Fix network ips

Fix ips

Fix ips again

PhiYerion committed:
Fix package.json

Update package.lock

PhiYerion committed:
Fix deploy

Minor deploy updates

Fix go test db port confict & deploy.sh

Fix deploy script

Fix deploy.sh

Fix deploy.sh

PhiYerion commented:
I'm unable to get any output when running the website other than a blank page

PhiYerion committed:
Remove /api from route now endpoint is api. ...

Update paths to exclude '/api'

Remove /api from workflows

PhiYerion committed:
Fix certs

Fix certs

Yetalon committed:
No more blank page

Yetalon committed:
Removed react-router-dom

Yetalon committed:
Merge branch 'bare-bones-frontend' of https://github.com/AppalachianCoding/rpg-app into bare-bones-frontend

Yetalon committed:
react-router fixed

Yetalon committed:
fixed package-lock

Yetalon committed:
Update userapihandler.js

PhiYerion reviewed (commented):
Changed implimented. Goood to go

PhiYerion committed:
Delete frontend/src/services/userapihandler.js

PhiYerion reviewed (approved):
approved
